src/Microsoft.IdentityModel.JsonWebTokens/JwtTokenUtilities.cs

```diff
@@ -429,6 +429,24 @@ private static long ParseTimeValue(JToken jToken, string claimName)
             throw LogHelper.LogExceptionMessage(new FormatException(LogHelper.FormatInvariant(LogMessages.IDX14300, LogHelper.MarkAsNonPII(claimName), jToken.ToString(), LogHelper.MarkAsNonPII(typeof(long)))));
         }
 
+        internal static string SafeLogJwtToken(object obj)
+        {
+            if (obj == null)
+                return string.Empty;
+
+            // not a string
+            if (!(obj is string token))
+                return obj.ToString();
```

Comment:
This doesn't seem safe at all. By doing this we are giving the impression of safety while we are not really doing any changes.

Reply:
We only mark string as Security artifact.

Reply:
What I mean here is that if I call your method I should be able to assume that your output is now safe. In the case of non strings you cannot guarantee that, you should throw here.

Reply:
Since this is really an exceptional case and should never happen you may consider throwing.

Reply:
return "UncognizedEncodedString"